### ISSUE TITLE
bug: false positive for DS0017 for arbitrary update command

### ISSUE THREAD

**Issue by Morl99** [aquasecurity/defsec#1256]
**Describe the bug**
There is a false positive for the following Dockerfile line
     
    RUN /bin/sh /scripts/someScript.sh update

    HIGH: The instruction 'RUN <package-manager> update' should always be followed by '<package-manager> install' in the same RUN statement.

**To Reproduce**

1. Write a Dockerfile with the above statement
2. run `trivy config`

**Expected behavior**
There should be no finding about this line
**Output of your tfsec command with --debug flag**
```
2023-03-28T15:16:41.815Z	INFO	Detected config files: 1
Dockerfile (dockerfile)
=======================
Tests: 21 (SUCCESSES: 20, FAILURES: 1, EXCEPTIONS: 0)
Failures: 1 (MEDIUM: 0, HIGH: 1, CRITICAL: 0)
HIGH: The instruction 'RUN <package-manager> update' should always be followed by '<package-manager> install' in the same RUN statement.
════════════════════════════════════════
The instruction 'RUN <package-manager> update' should always be followed by '<package-manager> install' in the same RUN statement.
See https://avd.aquasec.com/misconfig/ds017
────────────────────────────────────────
 Dockerfile:8
────────────────────────────────────────
   8 [ RUN /bin/sh /scripts/someScript.sh update
────────────────────────────────────────
```
**System Info**
/ $ trivy -v
Version: 0.38.3

**Comment by simar7** [comment]
Looks like a dupe of https://github.com/aquasecurity/defsec/issues/1235

**Comment by simar7** [comment]
ah you're right. Yeah this is a bug separate from the other one.

**Comment by simar7** [comment]
> I would suggest, that we switch the rule to a more targeted approach, and add all the known package managers into the rule, that would prevent false-positive matches. So the rule would only trigger, if either `update` or `--update` AND a package manager command like `apt-get`, `apk`, ... is found.
> 

Right, this makes sense as depending on the package manger options will be different. As you said we need to consider both of them to make a decision. At first we can start with `apt` and `apk` to begin with.

> What do you think? If that would be a way to move forward, we could prepare a first draft of a PR and start collecting package managers.

Sure! I'm happy to review any PRs you may have.

**Comment by Morl99** [comment]
I still plan to contribute.

**Comment by Morl99** [comment]
I have created a PR in which I limit the check to `yum`, `apt-get` and `apk`. I have added yum since it is widely used, and the current implementation already had yum specific install commands `groupinstall`. I also added a `RUN` statement to the existing `test_allowed` test, which covers the original use case of this issue. All the existing tests still pass, so I believe this PR is ready to be merged, if you have no objectsions @simar7

**Comment by Morl99** [comment]
After raising the PR, I was suprised to see, that I need to sign a CLA, this will take a while, because I have already completed our internal contribution process and was under the impression, that there is no CLA for this repo. As an improvement for future contributors, it would really help if the CLA would be included in the contribution guidelines.

We are only allowed to sign CLAs after a lawyer has looked into it.

**Comment by Morl99** [comment]
Just wanted to let you know, that our management is talking to yours to geht this impediment out of the way, but that it will take some time, since lawyers are involved :-)

**Comment by Morl99** [comment]
remove stale
